spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/SpannerConverters.java

```diff
@@ -112,16 +116,12 @@ public LocalDate convert(com.google.cloud.Date date) {
 	/**
 	 * A converter from {@link LocalDateTime} to the Spanner timestamp type.
 	 */
-	// @formatter:off
-	public static final Converter<LocalDateTime, Timestamp>
-			LOCAL_DATE_TIME_TIMESTAMP_CONVERTER = new Converter<LocalDateTime, Timestamp>() {
-		// @formatter:on
-		@Nullable
-		@Override
-		public Timestamp convert(LocalDateTime dateTime) {
-			return JAVA_TO_SPANNER_TIMESTAMP_CONVERTER.convert(java.sql.Timestamp.valueOf(dateTime));
-		}
-	};
+	public static final Converter<LocalDateTime, Timestamp> LOCAL_DATE_TIME_TIMESTAMP_CONVERTER =
+			CommitTimestamp.register(
+					new CommitTimestampSupplier<LocalDateTime>(
+							Value.COMMIT_TIMESTAMP.toSqlTimestamp().toLocalDateTime(),
+							SpannerConverters::toTimestamp
+					) { });
```

Comment:
Is it really necessary to use anonymous class?

Reply:
unfortunately, yes, it is necessary.
Today any `Converter` should be mandatory a class, not a lambda-function - types of "source" and "target" are resolved from the class as TypeArguments :(